File: app/src/main/java/com/confessionsearch/release1/data/notes/UpdateAsync.java
package com.confessionsearch.release1.data.notes;

import android.os.AsyncTask;

public class UpdateAsync extends AsyncTask<Notes, Void, Void> {

    private final NoteDao mNotesDao;

    public UpdateAsync(NoteDao dao) {
        mNotesDao = dao;
    }

    @Override
    protected Void doInBackground(Notes... notes) {
        mNotesDao.updateNotes(notes);
        return null;
    }
}

File: app/src/main/java/com/confessionsearch/release1/data/notes/UpdateAsync.kt
package com.confessionsearch.release1.data.notes

class UpdateAsync(private val mNotesDao: NoteDao) :
  CoroutinesAsyncTask<Notes?, Void?, Void?>("update")
{
  override fun doInBackground(vararg notes: Notes?): Void?
  {
    mNotesDao.updateNotes(*notes)
    return null
  }
}